src/modals/Validate.tsx

```diff
@@ -109,37 +101,32 @@ function Validate(props: ValidateProps) {
 
     const createSmartWallet = async () => {
         if (state.provider) {
-            dispatch({ type: 'set_loader', loader: true });
+            // dispatch({ type: 'set_loader', show: true });
             const smartWallet = await state.provider.generateSmartWallet(
                 Number(validate.address)
             );
             if (validateSmartWallets(smartWallet.address)) {
-                dispatch({ type: 'set_loader', loader: false });
                 return;
             }
-            const smartWalletWithBalance = await Utils.getSmartWalletBalance(
-                smartWallet,
-                state.token!
-            );
-            if (smartWalletWithBalance.deployed) {
-                const tempSmartWallets = [
-                    ...smartWallets,
-                    smartWalletWithBalance
-                ];
-                setSmartWallets(tempSmartWallets);
-                localStorage.setItem(
-                    Utils.getTransactionKey(state.chainId, state.account),
-                    JSON.stringify(tempSmartWallets)
-                );
-            } else {
-                setSmartWallets((prev) => [...prev, smartWalletWithBalance]);
+            const newSmartWallet = {
+                ...smartWallet,
+                tokenBalance: '0',
+                rbtcBalance: '0'
+            };
+            dispatch({
+                type: 'add_smart_wallet',
+                smartWallet: newSmartWallet
+            });
+            dispatch({ type: 'reload', reload: true });
+            if (smartWallet.deployed) {
+                Utils.addLocalSmartWallet(chainId, account, newSmartWallet);
             }
-            dispatch({ type: 'set_loader', loader: false });
+            // dispatch({ type: 'set_loader', show: false });
```

Comment:
Is it still necessary? If not I'd remove it.

Reply:
updated